fv3core/stencils/c_sw.py

```diff
@@ -206,24 +206,58 @@ def divergence_corner(
             * 0.5
             * (sin_sg4[0, -1] + sin_sg2)
         )
-        with horizontal(region[:, j_start], region[:, j_end + 1]):
-            uf = u * dyc * 0.5 * (sin_sg4[0, -1] + sin_sg2)
 
         vf = (
             (v - 0.25 * (ua[-1, 0, 0] + ua) * (cos_sg3[-1, 0] + cos_sg1))
             * dxc
             * 0.5
             * (sin_sg3[-1, 0] + sin_sg1)
         )
+
+        divg_d = (vf[0, -1, 0] - vf + uf[-1, 0, 0] - uf) * rarea_c
         with horizontal(region[i_start, :], region[i_end + 1, :]):
             vf = v * dxc * 0.5 * (sin_sg3[-1, 0] + sin_sg1)
+            vf1 = v[0, -1, 0] * dxc[0, -1] * 0.5 * (sin_sg3[-1, -1] + sin_sg1[0, -1])
+            uf1 = (
```

Comment:
It might be nice to in include a TODO of using a function with an argument with an offset, if you think that's a potential future feature. That would help reduce confusion later if someone tries to refactor this and gets overwhelmed with the long math statements.